yt_dlp/extractor/vk.py

```diff
@@ -701,3 +702,131 @@ def _real_extract(self, url):
         return self.playlist_result(
             entries, post_id, join_nonempty(uploader, f'Wall post {post_id}', delim=' - '),
             clean_html(get_element_by_class('wall_post_text', webpage)))
+
+
+class VKPlayBaseIE(InfoExtractor):
+    def _extract_initial_state(self, url, video_id):
+        webpage = self._download_webpage(url, video_id)
+        initial_state_json = self._search_regex(r'id="initial-state"[^>]*>([^<]+)</script>', webpage, 'initial_state')
+        initial_state = self._parse_json(initial_state_json, video_id, fatal=False)
+        return initial_state
+
+    def _parse_playurls(self, playurls, video_id):
+        formats = []
+        for playurl in playurls:
+            if not playurl.get('url', None):
+                continue
+            elif '.m3u8' in playurl['url']:
+                try:
+                    formats.extend(self._extract_m3u8_formats(playurl['url'], video_id))
+                except Exception:
+                    pass
+            else:
+                formats.append(playurl)
+        return formats
+
+
+class VKPlayIE(VKPlayBaseIE):
+    _VALID_URL = r'https?://vkplay\.live/(?P<username>[^/]+)/record/(?P<id>[a-f0-9\-]+)'
+    _TESTS = [{
+        'url': 'https://vkplay.live/zitsmann/record/f5e6e3b5-dc52-4d14-965d-0680dd2882da',
+        'info_dict': {
+            'id': 'f5e6e3b5-dc52-4d14-965d-0680dd2882da',
+            'ext': 'mp4',
+            'title': 'Atomic Heart (пробуем!) спасибо подписчику EKZO!',
+            'uploader': 'ZitsmanN',
+            'uploader_id': 13159830,
+            'release_timestamp': 1683461378,
+            'release_date': '20230507',
+            'thumbnail': r're:https://images.vkplay.live/public_video_stream/record/f5e6e3b5-dc52-4d14-965d-0680dd2882da/preview\?change_time=\d+',
+            'duration': 10608,
+            'view_count': int,
+            'like_count': int,
+            'categories': ['Atomic Heart'],
+        },
+        'params': {'skip_download': True},
```

Comment:
could do this, which will skip download and explain why we are doing so
```suggestion
        'params': {'skip_download': 'm3u8'},
```